pkg/api/vminfo.go

```diff
@@ -129,3 +151,63 @@ func ExtractVmInfo(vm *vmapi.VirtualMachine) (*VmInfo, error) {
 
 	return &info, nil
 }
+
+func (vm VmInfo) EqualScalingBounds(cmp VmInfo) bool {
+	return vm.Min() != cmp.Min() || vm.Max() != cmp.Max()
+}
+
+type scalingBounds struct {
+	Min *resourceBound `json:"min,omitempty"`
+	Max *resourceBound `json:"max,omitempty"`
```

Comment:
why need these be pointers?

Reply:
They're pointers so that we can distinguish between "value set to zero" and "value wasn't set"